### PR TITLE
CLI changes

### PR DESCRIPTION
**Pull request by mappum**
Made some minor UX improvements and fixes for the CLI.

### CODE REVIEW

**Comment by briantigerchow** [comment]
Does this PR fix a known sharness breakage?

```
# 1 known breakage(s) vanished; please update test(s)
```

https://build.protocol-dev.com/job/sharnessss/1061/console

**Comment by mappum** [comment]
>Does this PR fix a known sharness breakage?

I meant for it to, but it was still failing on my machine.

**Comment by briantigerchow** [comment]
> I meant for it to, but it was still failing on my machine.


What breakage is it meant to fix? I can try this on my machine.

**Comment by mappum** [comment]
It works when I run the commands manually, but this test fails for me in sharness: https://github.com/jbenet/go-ipfs/blob/23f810e187416262b158eebfc8a3817a973b5d34/test/sharness/t0090-get.sh#L71-L76

It's probably a problem with the tests but I couldn't figure it out.

File: test/sharness/t0090-get.sh [hunk 1]
```diff
@@ -54,7 +54,7 @@ test_expect_success "ipfs get -a output looks good" '
 test_expect_success "ipfs get -a archive output is valid" '
   tar -xf "$HASH".tar &&
   test_cmp $HASH data &&
-  rm $HASH.tar &&
+  rm "$HASH".tar &&
```
Comment (jbenet) [inline test/sharness/t0090-get.sh]: quote variables everywhere, that's probably what's going on. you have a ton more vars unquoted.

File: test/sharness/t0090-get.sh [hunk 2]
```diff
@@ -68,10 +68,8 @@ test_expect_success "ipfs get -a -C output looks good" '
   test_cmp expected actual
 '
 
-# TODO(mappum): fix this test. On osx I get:
-#     tar: Error opening archive: (Empty error message)
 test_expect_failure "gzipped tar archive output is valid" '
-  tar -xf "$HASH".tar.gz &&
+  tar -zxf $HASH.tar.gz &&
```
Comment (jbenet) [inline test/sharness/t0090-get.sh]: quotes

File: unixfs/tar/reader.go [hunk 3]
```diff
@@ -6,6 +6,7 @@ import (
 	"compress/gzip"
 	"io"
 	p "path"
+	"strings"
```
Comment (jbenet) [inline unixfs/tar/reader.go]: know its not part of this pr, but can you change `p` to something else? like `gopath` or something? `p` is not very clear.

File: core/commands/get.go [hunk 4]
```diff
@@ -76,7 +77,7 @@ may also specify the level of compression by specifying '-l=<1-9>'.
 
 		outPath, _, _ := req.Option("output").String()
 		if len(outPath) == 0 {
-			outPath = req.Arguments()[0]
+			_, outPath = p.Split(req.Arguments()[0])
```
Comment (jbenet) [inline core/commands/get.go]: might want to pass it through `path.Clean`

**Comment by jbenet** [comment]
- [ ] quote all variables in sharness
- [ ] `path.Clean`
- [ ] `p` something else

**Comment by jbenet** [comment]
LGTM!
